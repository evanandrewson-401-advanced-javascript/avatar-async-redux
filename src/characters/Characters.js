import React from 'react';
import Character from './Character';
import PropTypes from 'prop-types';

const Characters = ({ items }) => {
  const elements = items.map((item, i) => {
    return <li key={item._id || i}><Character {...item} /></li>;
  });
  return (
    <ul>
      {elements}
    </ul>
  );
};

Characters.propTypes = {
  toDetailPage: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired
};

export default Characters;
