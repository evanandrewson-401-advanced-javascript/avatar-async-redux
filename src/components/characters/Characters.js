import React from 'react';
import Character from './Character';
import PropTypes from 'prop-types';

const Characters = ({ items }) => {
  const elements = items.map((item, i) => {
    return <li key={item._id || i}><Character {...item} /></li>;
  });
  return (
    <ul>
      {elements}
    </ul>
  );
};

Characters.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    photoUrl: PropTypes.string.isRequired
  })).isRequired
};

export default Characters;
