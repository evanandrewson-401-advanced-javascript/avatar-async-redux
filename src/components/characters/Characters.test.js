import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

const characters = [{
  _id: '5cf5679a915ecad153ab68c8',
  allies: [
    'Royal Earthbender Guards'
  ],
  enemies: [
    'Chin'
  ],
  photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
  name: '46th Earth King',
  affiliation: ' Earth Kingdom Earth Kingdom Royal Family',
}];

describe('Characters component', () => {
  it('renders Characters', () => {
    const wrapper = shallow(<Characters items={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
