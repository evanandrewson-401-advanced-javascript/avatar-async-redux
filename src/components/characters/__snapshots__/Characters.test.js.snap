// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders Characters 1`] = `
<ul>
  <li
    key="5cf5679a915ecad153ab68c8"
  >
    <Character
      _id="5cf5679a915ecad153ab68c8"
      affiliation=" Earth Kingdom Earth Kingdom Royal Family"
      allies={
        Array [
          "Royal Earthbender Guards",
        ]
      }
      enemies={
        Array [
          "Chin",
        ]
      }
      name="46th Earth King"
      photoUrl="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
    />
  </li>
</ul>
`;
